src/service/github.oauth.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as express from "express";
+import { Server } from "http";
+import fetch from "node-fetch";
+import { URLSearchParams } from "url";
+import { state } from "../state";
+
+export class GitHubOAuthService {
+  public app: express.Express;
+  public server: Server;
+
+  constructor(public port: number) {
+    this.app = express();
+    this.app.use(express.json(), express.urlencoded({ extended: false }));
+  }
+
+  public async StartProcess() {
+    this.server = this.app.listen(this.port);
+    this.app.get("/callback", async (req, res) => {
+      const params = new URLSearchParams(
+        await (await this.getToken(req.param("code"))).text()
+      );
+
+      res.send("<script>window.close();</script>");
+      this.server.close();
+
+      const token = params.get("access_token");
+      this.saveToken(token);
+
+      const user = await this.getUser(token);
+
+      const gists = await this.getGists(token, user);
+      state.commons.webviewService.OpenGistSelectionpage(gists);
+    });
+  }
+
+  public getToken(code: string) {
+    const params = new URLSearchParams();
+    params.append("client_id", "cfd96460d8b110e2351b");
+    params.append("client_secret", "ed46bd3a0f736e0da57308e86ca5fa3cf8688582");
+    params.append("code", code);
+
+    return fetch("https://github.com/login/oauth/access_token", {
```

Comment:
`https://github.com/login/oauth/access_token`

What about github enterprise login defined in Global Settings.
We need to create url depending on default github or user github enterprise login and make api calls over there instead of hard coding here.

Reply:
Done